x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/common_attributes.ts

```diff
@@ -286,3 +292,21 @@ export const InvestigationFields = t.exact(
     field_names: NonEmptyArray(NonEmptyString),
   })
 );
+
+// 8.10.x is mapped as an array of strings
+export type LegacyInvestigationFields = t.TypeOf<typeof LegacyInvestigationFields>;
+export const LegacyInvestigationFields = NonEmptyStringArray;
```

Comment:
As @maximpn found out, this is an incorrect type. The schema should be either of these:

```ts
export const LegacyInvestigationFields = NonEmptyArray(NonEmptyString);
export const LegacyInvestigationFields = t.array(NonEmptyString);
export const LegacyInvestigationFields = NonEmptyArray(t.string);
export const LegacyInvestigationFields = t.array(t.string);
```

Probably the latter one will give the most flexibility. We would accept any kinds of arrays: empty and non-empty, also containing empty strings potentially, and will have to handle these edge cases in the migration function on the server side.